### PR TITLE
Feature/mdr 111

### PR DESCRIPTION
**Pull request by costezki**
please review

### CODE REVIEW

File: .readthedocs.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+
+# Required
+version: 2
+
+# Build documentation in the docs/ directory with Sphinx
+sphinx:
+  configuration: docs/conf.py
+
+# Build documentation with MkDocs
```
Comment (cosletmihai) [inline .readthedocs.yml:11]: let's not keep unused comments

File: README.md [hunk 2]
```diff
@@ -1,3 +1,20 @@
...
+```
+
+```jinja2
+{% set content, error =
+    from_endpoint(endpoint).with_query(query_string).fetch_tabular() %}
+content:  {{ content }} \n
+error: {{ error }} \n
+```
+
+[Read the docs here](https://eds4jinja2.readthedocs.io/en/feature-mdr-111/)
```
Comment (cosletmihai) [inline README.md:20]: why the `/feature-mdr-111`?

File: docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+# documentation root, use os.path.abspath to make it absolute, like shown here.
+#
+import os
+import sys
+
+sys.path.insert(0, os.path.abspath('..'))
+
+# -- Project information -----------------------------------------------------
+# The full version, including alpha/beta/rc tags
+release = "0.0.17"
```
Comment (cosletmihai) [inline docs/conf.py:20]: the version [here](https://pypi.org/project/edt4jinja2/) mentions version `0.0.18`

File: docs/index.rst [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+3. render the template with no context. The context is dynamically generated during the template rendering.
+
+.. code-block:: py3
+
+    rendered_text = template.render()
+
+**To query a SPARQL endpoint**
+
+Repeat the steps from the example above to instantiate the environment and render the template. But adapt teh template as follows.
```
Comment (cosletmihai) [inline docs/index.rst:104]: adapt **the**

File: docs/index.rst [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+2. modify the template file 'mytemplate.txt' to looks like this
```
Comment (cosletmihai) [inline docs/index.rst:106]: to **look**

File: docs/index.rst [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+    {% set query_string = "select * where {?s ?p ?o} limit 10" %}
+    {% set content, error = from_endpoint(endpoint).with_query(query_string).fetch_tree() %} \n
+    content: {{ content }}\n
+    error: {{ error }}\n
+
+So, what are the benefits?
+####################################################
+
+* your python code is agnostic of what data teh template displays
```
Comment (cosletmihai) [inline docs/index.rst:118]: data **the**

File: eds4jinja2/adapters/sparql_ds.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+class SPARQLEndpointDataSource(DataSource):
+    """
+        Fetches data from SPARQL endpoint. Can be used either with a SPARQL query or a URI to be described.
+
+        To query a SPARQL endpoint and get the results as *dict* object
+
+        >>> ds = SPARQLEndpointDataSource(sparql_endpoint_url)
+        >>> dict_object = ds.with_query(sparql_query_text)._fetch_tree()
+
+        unpack the content and error for a fail safe safe fetching
```
Comment (cosletmihai) [inline eds4jinja2/adapters/sparql_ds.py:29]: is the double **safe** intentional?

File: tests/unit/test_sparql_ds.py [hunk 8]
```diff
@@ -4,3 +4,67 @@
...
+
+from eds4jinja2.adapters.sparql_ds import SPARQLEndpointDataSource
+from tests import ENDPOINT_REMOTE_CORRECT, DUMMY_DESCRIBE_URI, SPO_LIMIT_10, WRONG_SPO_LIMIT_10, \
+    QUERY_LONGER_THAN_2048KB, ENDPOINT_LOCAL_CORRECT, ENDPOINT_INEXISTENT_SERVER, DUMMY_DESCRIBE_URI_GRAPH
+
+
+def test_connect_to_remote_endpoint():
+    fds = SPARQLEndpointDataSource(ENDPOINT_REMOTE_CORRECT)
+    response_text = str(fds.with_uri(DUMMY_DESCRIBE_URI)._fetch_tree())
+    assert len(response_text) > 2000
```
Comment (cosletmihai) [inline tests/unit/test_sparql_ds.py:20]: looks like a too broad of an `assert`

File: tests/unit/test_sparql_ds.py [hunk 9]
```diff
@@ -4,3 +4,67 @@
...
+    assert "results" in response_text
+
+
+def test_connect_to_endpoint_fails():
+    with pytest.raises(ValueError):
+        fds = SPARQLEndpointDataSource("")
+        print(fds._fetch_tree())
```
Comment (cosletmihai) [inline tests/unit/test_sparql_ds.py:27]: why are you printing these?

File: README.md (hunk 2 above)
Comment (costezki) [reply README.md:20]: this is a test that it works. 
Will change now to the latest now

File: docs/conf.py (hunk 3 above)
Comment (costezki) [reply docs/conf.py:20]: implemented a unifying mechanism here

File: docs/index.rst (hunk 4 above)
Comment (costezki) [reply docs/index.rst:104]: fixed

File: docs/index.rst (hunk 5 above)
Comment (costezki) [reply docs/index.rst:106]: fixed

File: docs/index.rst (hunk 6 above)
Comment (costezki) [reply docs/index.rst:118]: fixed

File: eds4jinja2/adapters/sparql_ds.py (hunk 7 above)
Comment (costezki) [reply eds4jinja2/adapters/sparql_ds.py:29]: error: fixed

File: tests/unit/test_sparql_ds.py (hunk 8 above)
Comment (costezki) [reply tests/unit/test_sparql_ds.py:20]: yeah, just wanted to know that the connection is performed successfully.

File: tests/unit/test_sparql_ds.py (hunk 9 above)
Comment (costezki) [reply tests/unit/test_sparql_ds.py:27]: removed

**Pull request merged**